optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -0,0 +1,385 @@
+        cache_dir: Optional[str] = None,
+        **kwargs,
+    ):
+        # Create local save dir in cache dir
+        save_dir = Path(save_dir).joinpath(model_id)
+        save_dir.mkdir(parents=True, exist_ok=True)
+        kwargs["model_save_dir"] = save_dir
+
+        task = "seq2seq-lm"
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        model = FeaturesManager.get_model_from_feature(task, model_id)
+        _, model_onnx_config = FeaturesManager.check_supported_model_or_raise(model, feature=task)
+        onnx_config = model_onnx_config(model.config)
+        onnx_opset = onnx_config.default_onnx_opset
+        onnx_config_encoder = EncoderOnnxConfig(model.config, task="default")
+        onnx_config_decoder = DecoderOnnxConfig(model.config, task=task, use_past=False)
+        onnx_config_decoder_with_past = DecoderOnnxConfig(model.config, task=task, use_past=True)
+
+        # Extract the encoder for ONNX export
+        encoder = model.get_encoder()
+        # Concatenate the decoder with the language model head for ONNX export
+        decoder_with_lm_head = _DecoderWithLMhead(model)
+
+        # Export the encoder
+        export(
+            preprocessor=tokenizer,
+            model=encoder,
+            config=onnx_config_encoder,
+            opset=onnx_opset,
+            output=save_dir.joinpath(ONNX_ENCODER_NAME),
+        )
+
+        # Export the decoder without the past key values
+        export(
+            preprocessor=tokenizer,
+            model=decoder_with_lm_head,
+            config=onnx_config_decoder,
+            opset=onnx_opset,
+            output=save_dir.joinpath(ONNX_DECODER_NAME),
+        )
+
+        # Export the decoder with the past key values
+        export(
+            preprocessor=tokenizer,
+            model=decoder_with_lm_head,
+            config=onnx_config_decoder_with_past,
+            opset=onnx_opset,
+            output=save_dir.joinpath(ONNX_DECODER_WITH_PAST_NAME),
+        )
+
+        kwargs["config"] = model.config.__dict__
+        return cls._from_pretrained(save_dir.as_posix(), **kwargs)
+
+    def generate(self, *args, **kwargs):
+        return self.model.generate(*args, **kwargs)
+
+    def forward(self, *args, **kwargs):
+        return self.model.forward(*args, **kwargs)
+
+
+class ORTEncoder(torch.nn.Module):
+    def __init__(self, encoder: onnxruntime.InferenceSession):
+        super().__init__()
+        self.encoder = encoder
+        self.main_input_name = "input_ids"
+        self.output_names = {output_key.name: idx for idx, output_key in enumerate(self.encoder.get_outputs())}
+
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        **kwargs,
+    ) -> BaseModelOutput:
+
+        onnx_inputs = {
+            "input_ids": input_ids.cpu().detach().numpy(),
+            "attention_mask": attention_mask.cpu().detach().numpy(),
```

Comment:
I would use `np.ndarray` at this stage and not have any reference to PyTorch

Reply:
The utilization of generate makes it not feasible at the moment : currently the inputs as well as the outputs of our encoder needs to be `torch.<X>Tensor`